modules/icgc-argo-workflows/payload/alignment/resources/usr/bin/main.py

```diff
@@ -0,0 +1,220 @@
+            'inputs': [
+                {
+                    'analysis_type': 'sequencing_experiment',
+                    'input_analysis_id': seq_experiment_analysis_dict.get('analysisId')
+                }
+            ]
+        },
+        'files': [],
+        'samples': get_sample_info(seq_experiment_analysis_dict.get('samples')),
+        'experiment': {},
+        'read_group_count': seq_experiment_analysis_dict.get('read_group_count'),
+        'read_groups': seq_experiment_analysis_dict.get('read_groups')
+    }
+    if args.genome_annotation:
+        payload['workflow']['genome_annotation'] = args.genome_annotation
+
+    # pass `info` dict from seq_experiment payload to new payload
+    if 'info' in seq_experiment_analysis_dict and isinstance(seq_experiment_analysis_dict['info'], dict):
+        payload['info'] = seq_experiment_analysis_dict['info']
+    else:
+        payload.pop('info')
+
+    payload['experiment'].update(seq_experiment_analysis_dict.get('experiment', {}))
+
+    if 'library_strategy' in payload['experiment']:
+        experimental_strategy = payload['experiment'].pop('library_strategy')
+        payload['experiment']['experimental_strategy'] = experimental_strategy
+
+    # get inputs from read_group_ubam_analysis
+    for ubam_analysis in args.read_group_ubam_analysis:
+        with open(ubam_analysis, 'r') as f:
+            ubam_analysis_dict = json.load(f)
+
+        payload['workflow']['inputs'].append(
+            {
+                'analysis_type': 'read_group_ubam',
+                'input_analysis_id': ubam_analysis_dict.get('analysisId')
+            }
+        )
+
+    # get number of read groups from aligned seq file
+    # aligned_file = [ f for f in args.files_to_upload if (f.endswith('.bam') or f.endswith('.cram')) ][0]
+    rg_count = args.read_group_count
+
+    # get file of the payload
+    date_str = date.today().strftime("%Y%m%d")
+    for f in args.files_to_upload:
+        renamed_file = rename_file(f, payload, rg_count, seq_experiment_analysis_dict['samples'], date_str)
+        payload['files'].append(get_files_info(renamed_file))
+
+    with open("%s.%s.payload.json" % (str(uuid.uuid4()), args.wf_name.replace(" ","_")), 'w') as f:
+        f.write(json.dumps(payload, indent=2))
+
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Tool: payload-rnaseq-alignment')
+    parser.add_argument("-f", "--files_to_upload", dest="files_to_upload", type=str, required=True,
+                        nargs="+", help="Aligned reads files to upload")
+    parser.add_argument("-a", "--seq_experiment_analysis", dest="seq_experiment_analysis", required=True,
+                        help="Input analysis for sequencing experiment", type=str)
+    parser.add_argument("-u", "--read_group_ubam_analysis", dest="read_group_ubam_analysis", default=[],
+                        help="Input payloads for the analysis", type=str, nargs='+')
```

Comment:
The option can be removed. We don't have `read_group_ubam_analysis`.

Reply:
removed